### PR TITLE
feat(timeline): token migration and update shim

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/timeline/_timeline.clarity.scss [hunk 1]
```diff
@@ -190,30 +260,42 @@
...
+        clr-timeline-step-internal-margin,
+        timeline-variables.$clr-timeline-vertical-step-body-left-padding,
+        variables.$clr-use-custom-properties
+      );
+      @include mixins.css-var(
+        margin-top,
+        clr-timeline-internal-margin-0,
+        timeline-variables.$clr-timeline-vertical-step-body-top-padding,
+        variables.$clr-use-custom-properties
+      );
```
Comment (dtsanevmw) [inline projects/angular/src/timeline/_timeline.clarity.scss:298]: I would expose general margin token and use it like that, this `*-margin-0` doesn't look right.
```suggestion
      @include mixins.css-var(
        margin,
        clr-timeline-step-internal-margin,
        timeline-variables.$clr-timeline-vertical-step-body-top-padding 0 0 timeline-variables.$clr-timeline-vertical-step-body-left-padding,
        variables.$clr-use-custom-properties
      );
```

File: projects/angular/src/timeline/_timeline.clarity.scss [hunk 2]
```diff
@@ -90,16 +111,24 @@
...
       variables.$clr-use-custom-properties
     );
-    font-size: timeline-variables.$clr-timeline-step-header-font-size;
-    line-height: timeline-variables.$clr-timeline-step-header-line-height;
-    white-space: nowrap;
-    margin-bottom: timeline-variables.$clr-timeline-step-header-padding-bottom;
+
+    font-weight: var(--clr-timeline-step-header-font-weight, inherit);
+    font-size: var(--clr-timeline-step-header-font-size, timeline-variables.$clr-timeline-step-header-font-size);
+    line-height: var(--clr-timeline-step-header-line-height, timeline-variables.$clr-timeline-step-header-line-height);
```
Comment (dtsanevmw) [inline projects/angular/src/timeline/_timeline.clarity.scss:124]: Maybe use the mixin here?

**Review by dtsanevmw — CHANGES_REQUESTED** [review]
Two suggestions in the comments plus you could add the newly added token from the shim in _properties.

File: projects/angular/src/timeline/_timeline.clarity.scss (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/timeline/_timeline.clarity.scss]: corrected in commit 0a5d649

File: projects/angular/src/timeline/_timeline.clarity.scss [hunk 3]
```diff
@@ -167,19 +245,39 @@
...
-            timeline-variables.$clr-timeline-vertical-step-body-top-padding
+        @include mixins.css-var(
+          width,
+          clr-timeline-line-width,
+          timeline-variables.$clr-timeline-line-thickness,
+          variables.$clr-use-custom-properties
+        );
+        @include mixins.css-var(
+          height,
+          clr-timeline-vertical-line-lenght,
```
Comment (dtsanevmw) [inline projects/angular/src/timeline/_timeline.clarity.scss:267]: nit: typo
```suggestion
          clr-timeline-vertical-line-length,
```

Comment (valentin-mladenov) [reply projects/angular/src/timeline/_timeline.clarity.scss]: Typo fixed in 8d8e5e1

File: projects/ui/src/shim.cds-core.scss [hunk 4]
```diff
@@ -870,16 +870,41 @@
...
+  --clr-timeline-step-margin: var(--cds-global-space-5);
+  --clr-timeline-step-internal-margin: var(--cds-global-space-6);
+  --clr-timeline-line-width: var(--cds-global-space-2);
+  --clr-timeline-horizontal-line-length: calc(100% - calc(var(--clr-timeline-icon-size) - var(--clr-timeline-step-internal-margin)));
+  --clr-timeline-vertical-line-position: translate(
+    calc(-0.5 * calc(var(--clr-timeline-icon-size) + var(--clr-timeline-line-width)) - var(--clr-timeline-step-internal-margin)),
+    calc(var(--clr-timeline-icon-size) - var(--clr-timeline-line-width))
+  );
+
+  --clr-timeline-vertical-line-lenght: calc(100% - calc(0.5 * var(--clr-timeline-icon-size) + var(--clr-timeline-line-width)));
```
[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/ui/src/shim.cds-core.scss]: Typo fixed in 8d8e5e1

**Pull request merged**